### PR TITLE
Receive and send audio messages

### LINKED ISSUE #19: Feature request: receive and send audio message
Hi, Thanks for the nice OA puppet. I'm interested in receiving and sending audio messages with wechat users through OA. Currently the puppet.messageSendFile() method throws an 'unsupported file exception' when trying to send SILK encoded audio files, or .mp3, and the 'EventMessagePayload' also does not receive audio input from the user. 
Would appreciate it if we can add this new feature, please instruct how to do it. Thanks!

### PR DESCRIPTION
**Pull request by zhangfand**
Fixes #19 

The change adds support to OA puppet to receive and send audio files from / to Wechat,
and add another example that "echo" text or audio messages.

The implementation is largely constructed per the detailed instructions in #19 (❤️ ) .
There are a few catches:
1. Have to make `messageSend` public and use it in the example instead of using `messageSendFile`. The main reason is `mime` package can not infer MIME type of `amr` file extension.
2. How Wechat `/get/media` API handles audio file is not well documented. From experiments, it seems to always return the audio file. I've documented the assumed format in the code.

### CODE REVIEW

**Review by huan — CHANGES_REQUESTED** [review]
Your PR looks great, it has implemented most of the code for supporting the audio message for the OA puppet!

However, there has two parts of the code that we need to improve before we can merge it.

Please see my review at https://github.com/wechaty/wechaty-puppet-official-account/issues/19#issuecomment-775365848

File: examples/echo.ts
Comment (huan) [inline examples/echo.ts:82]: Could you explain what this code is doing?

In my understanding, this is the official account send a message to itself?

File: examples/echo.ts
Comment (huan) [inline examples/echo.ts:106]: The `payload` might not a file type.

What if a user just sends a text to the official account?

Need to be checked.

**Review by huan — CHANGES_REQUESTED** [review]
Thanks for the update!

Please see my reviews for some minor questions.

And for better unit testing, I believe the best practice is to build a Mock HTTP Server to interact with the code, and make sure our code is behavior as expected.

File: examples/echo.ts
Comment (zhangfand) [reply examples/echo.ts]: This code is a copy-pasta from `ding-dong-bot.ts`. I end up with deleting it since it is not really what I want to demonstrate in this example.

File: examples/echo.ts
Comment (zhangfand) [reply examples/echo.ts]: See the comment below. TL;DR I deleted it in the newest commit.

File: examples/echo.ts
Comment (zhangfand) [reply examples/echo.ts]: Good call. I updated the code to handle different types of messages.

**Comment by zhangfand** [comment]
Replied to questions in the code review.

**Review by huan — CHANGES_REQUESTED** [review]
I believe this PR will be ready to merge after the CI turns green.

**Comment by wj-Mcat** [comment]
Kindly ping @zhangfand

**Comment by zhangfand** [comment]
The newest `mime` package adds extentions mapping for `audio/amr`, which causes initialization failure in file-box (see failed test log).

huan/file-box#53 should fix the issue.

**Comment by huan** [comment]
Dove. Thank you very much for keeping the modules update!

**Review by huan — APPROVED** [review]
LGTM

**Comment by huan** [comment]
Thank you very much for your contribution!

## You are welcome to join [Wechaty Contributor Program](https://wechaty.js.org/docs/contributor-program/)

### 1. Join Wechaty Organization

> You've invited Zhangfan to Wechaty! They'll be receiving an email shortly. They can also visit https://github.com/wechaty to accept the invitation.

I have invited you to join our Wechaty GitHub Organization, please accept it by following the above message.

### 2. Update Your Wechaty Contributor Profile

1. Please open [Contributor Hall of Fame](https://docs.google.com/spreadsheets/d/1XcDoIczyIclqXP1p90Sz7S0n4Q22xjFzJmjPFlU2g1E/edit#gid=0) and add yourself to the end of the list, so that other contributors will know you better!
1. Please add yourself to our Website [Contributors](https://wechaty.js.org/contributors/) by [creating a PR](https://github.com/wechaty/wechaty.js.org/tree/master/jekyll/_contributors) and refer to this PR link as well.

### 3. Join The Contributor Only WeChat Room

We also have a WeChat room for contributors only which can discuss Wechaty at a deeper level, you are welcome to join and if you are interested.

Please add @lijiarui wechat: _ruirui_0914_ and send her this pr link. She will invite you into Wechaty Contributor Room

Cheers!

**Pull request merged**
